File: sorting/selectionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

//Start Denotes the index from where the unsorted Region Starts
int findMinIndex(vector<int>arr, int n, int start){
   int  min_index = start; // first element of the unsorted region

   for(int i = start +1; i < n; i++){
       if(arr[i] < arr[min_index]){
           // then we have a new Minimum.....
           min_index = i;
       }
   }
    return min_index;
}

vector<int> selectionSort(vector<int> arr, int n){
    for(int i=0; i < n-1; i++){
        int min_index = findMinIndex(arr, n, i);
        int temp = arr[i];
        arr[i] = arr[min_index];
        arr[min_index] = temp;

         

    }

    for(int i = 0 ; i <n; i++){
        cout << arr[i] << " ";
    }
    cout << endl ;

    return arr;
}

int main(){
    
    int n = 8;
    vector<int> arr = { 5,2,6,7,2,1,0,3 };
    selectionSort(arr, n);
    return 0;
}

File: sorting/selectionSort.js

let arr = [5,2,6,7,2,1,0];
console.log(arr);

console.log(selectionSort(arr));

console.log("Hii")

//Start Denotes the index from where the unsorted Region Starts
function findMinIndex(arr, start){
    let min_indx = start; // We will assume that the first element of unsorted Region
    for(let i = start+1; i , arr.length; i++){
        if(arr[i] < arr[min_indx]){
            //then we have found a new minimum....
            min_indx = i;
        }
    }
    return min_indx;
}

function selectionSort(arr){
    for(let i = 0; i < arr.length-1; i++){
        let min_indx = findMinIndex(arr, i);
        if( i != min_indx){
            let temp = arr[i];
            arr[i] = arr[min_indx];
            arr[min_indx] = temp;
        }
    }
    console.log("HIi")
}